discovery/source/DiscoveryClient.cpp

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+
+* This file is generated
+*/
+#include <aws/discovery/DiscoveryClient.h>
+
+#include <aws/crt/Types.h>
+#include <aws/crt/io/TlsOptions.h>
+#include <aws/crt/io/Uri.h>
+
+namespace Aws
+{
+    namespace Discovery
+    {
+        DiscoveryClientConfig::DiscoveryClientConfig() noexcept
+            : allocator(Crt::DefaultAllocator()), bootstrap(nullptr), tlsContext(nullptr), socketOptions(nullptr),
+              region("us-east-1"), maxConnections(2)
+        {
+        }
+
+        DiscoveryClient::DiscoveryClient(const Aws::Discovery::DiscoveryClientConfig &clientConfig) noexcept
+        {
+            assert(clientConfig.tlsContext);
+            assert(clientConfig.bootstrap);
+            assert(clientConfig.socketOptions);
+
+            m_allocator = clientConfig.allocator;
+
+            Crt::StringStream ss;
+            ss << "greengrass-ats.iot." << clientConfig.region << ".amazonaws.com";
+
+            Crt::Io::TlsConnectionOptions tlsConnectionOptions = clientConfig.tlsContext->NewConnectionOptions();
+            uint16_t port = 443;
+
+            if (Crt::Io::TlsContextOptions::IsAlpnSupported())
+            {
+                tlsConnectionOptions.SetAlpnList("x-amzn-http-ca");
```

Comment:
Does this need follow-up changes in aws-c-http to work?

Reply:
yep, and done